src/Runner.js

```diff
@@ -48,6 +49,31 @@ function concatAll(arrays) {
   );
 }
 
+function getStdin() {
+  return new Promise(resolve => {
+    const stdin = process.stdin;
+    if (!stdin || stdin.isTTY) {
+      resolve(null);
+    } else {
+      // parse stdin to string content
+      let rst = '';
+
+      stdin.setEncoding('utf8');
+
+      stdin.on('readable', () => {
+        let chunk;
+        while ((chunk = stdin.read())) {
+          rst += chunk;
+        }
+      });
+
+      stdin.on('end', () => {
+        resolve(rst);
+      });
+    }
+  });
+}
+
```

Comment:
Instead of buffering the entire file from `stdin` and then writing it to disk (only to be re-read again when we process it), can you instead change this to write `process.stdin` directly to an open handle to a temporary file (e.g. using `fs.createWritesStream` and `process.stdin.pipe`)?